hack/e2e.sh

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script replaces hack/ginkgo-e2e.sh when not using a cloud provider.
+#
+# It always runs e2e tests without provider=skeleton and reads cluster info only
+# from KUBECONFIG, with no special GCP setup code and no dependency on cluster/
+#
+# It does not pass any cloud provider related flags and is not compatible with
+# ginkgo-e2e.sh, please use ginkgo-e2e.sh when testing with the existing cloud
+# provider support and tests that require this.
+#
+#
+# Usage: `hack/e2e.sh`.
+#
+# Make sure the current KUBECONFIG points to the desired cluster to test.
+#
+# You also need to build e2e.test and ginkgo, and you should build kubectl though
+# it can fall back to the system kubectl.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Options
+################################################################################
+# GINKGO_PARALLEL: set to 'y' to run tests in parallel
+GINKGO_PARALLEL=${GINKGO_PARALLEL:-n}
+# GINKGO_SILENCE_SKIPS: set to 'n' to see S character for each skipped test
+GINKGO_SILENCE_SKIPS=${GINKGO_SILENCE_SKIPS:-y}
+# GINKGO_FORCE_NEWLINES: set to 'y' to print a newline after each S or o character
+GINKGO_FORCE_NEWLINES=${GINKGO_FORCE_NEWLINES:-$( if [ "${CI:-false}" = "true" ]; then echo "y"; else echo "n"; fi )}
+# GINKGO_NO_COLOR: If 'y', Ginkgo's reporter will not use escape sequence to color output.
+#
+# Since Kubernetes 1.25, the default is to use colors only when connected to
+# a terminal. That is the right choice for all Prow jobs (The UI doesn't
+# render them properly).
+GINKGO_NO_COLOR=${GINKGO_NO_COLOR:-$(if [ -t 2 ]; then echo n; else echo y; fi)}
+# E2E_TEST_DEBUG_TOOL: If set, the command executed will be:
+# - `dlv exec` if set to "delve"
```

Comment:
As I proposed in [ci: allow devs using delve to debug e2e locally without gce #133253](https://github.com/kubernetes/kubernetes/pull/133253), using port listening method should be better than mere `exec` flag

Reply:
Sure, we can iterate on that, the bulk of this script is a direct copy from the previous script as a starting point (see the first commit)

This would be a good change to add in a later commit